### PR TITLE
Split & parallelize tests

### PR DESCRIPTION
**Pull request by benoitvidis**
This PR splits the tests to run them in parallel.

Other changes:
- do not rely on manually pushed docker containers anymore (please warn if/when merging this pr to clean docker hub)
- upgrade docker images to last versions (fedora 28, ubuntu bionic, debian stretch).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,43 @@
...
+  - chmod 600 ~/.ssh/id_rsa
+  - sudo sysctl -w vm.max_map_count=262144
+
+stages:
+  - name: test
+
+jobs:
+  include:
+    - stage: test
+      name: fedora 28
```
Comment (alexandrebouthinon) [inline .travis.yml]: Nitpicking: In the other repositories Travis configuration, I capitalize stage name.

File: .travis.yml [hunk 2]
```diff
@@ -1,14 +1,43 @@
...
+      env:
+      script:
+        - test/test-setup.sh fedora
+
+    - stage: test
+      name: debian stretch
```
Comment (alexandrebouthinon) [inline .travis.yml]: same here

File: .travis.yml [hunk 3]
```diff
@@ -1,14 +1,43 @@
...
+      env:
+      script:
+      - test/test-setup.sh debian
+
+    - stage: test
+      name: ubuntu bionic
```
Comment (alexandrebouthinon) [inline .travis.yml]: same here

File: .travis.yml [hunk 4]
```diff
@@ -1,14 +1,43 @@
...
+      env:
+      script:
+        - test/test-setup.sh ubuntu
+
+    - stage: test
+      name: macosx
```
Comment (alexandrebouthinon) [inline .travis.yml]: Same here

**Review by alexandrebouthinon — APPROVED** [review]

File: .travis.yml (hunk 1 above)
Comment (alexandrebouthinon) [inline .travis.yml]:
```suggestion
      name: Fedora - 28
```

File: .travis.yml (hunk 2 above)
Comment (alexandrebouthinon) [inline .travis.yml]:
```suggestion
      name: Debian Stretch - 9.0
```

File: .travis.yml (hunk 3 above)
Comment (alexandrebouthinon) [inline .travis.yml]:
```suggestion
      name: Ubuntu Bionic - 18.04
```

File: .travis.yml (hunk 4 above)
Comment (alexandrebouthinon) [inline .travis.yml]:
```suggestion
      name: Mac OSX
```

**Review by alexandrebouthinon — APPROVED** [review]

**Review by scottinet — APPROVED** [review]

**Pull request merged**